From f5163d08d09a7f295012f70cf0193d3fbc879eec Mon Sep 17 00:00:00 2001
From: Mathias Kresin <dev@kresin.me>
Date: Sun, 3 Jan 2021 23:38:49 +0100
Subject: [PATCH] generic: ar8216: add kernel 5.5+ compatiblity

of_get_phy_mode() returns an error, or 0 on success, and pass a
pointer, of type phy_interface_t, where the phy mode should be
stored now.

Signed-off-by: Mathias Kresin <dev@kresin.me>
---
 target/linux/generic/files/drivers/net/phy/ar8216.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/target/linux/generic/files/drivers/net/phy/ar8216.c b/target/linux/generic/files/drivers/net/phy/ar8216.c
index acfa0ebecd..3bcab832fd 100644
--- a/target/linux/generic/files/drivers/net/phy/ar8216.c
+++ b/target/linux/generic/files/drivers/net/phy/ar8216.c
@@ -891,7 +891,12 @@ ar8216_phy_write(struct ar8xxx_priv *priv, int addr, int regnum, u16 val)
 static int
 ar8229_hw_init(struct ar8xxx_priv *priv)
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,5,0)
+	phy_interface_t phy_if_mode;
+	int err;
+#else
 	int phy_if_mode;
+#endif
 
 	if (priv->initialized)
 		return 0;
@@ -899,7 +904,13 @@ ar8229_hw_init(struct ar8xxx_priv *priv)
 	ar8xxx_write(priv, AR8216_REG_CTRL, AR8216_CTRL_RESET);
 	ar8xxx_reg_wait(priv, AR8216_REG_CTRL, AR8216_CTRL_RESET, 0, 1000);
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,5,0)
+	err = of_get_phy_mode(priv->pdev->of_node, &phy_if_mode);
+	if (err)
+		pr_err("ar8229: no mii mode set\n");
+#else
 	phy_if_mode = of_get_phy_mode(priv->pdev->of_node);
+#endif
 
 	if (phy_if_mode == PHY_INTERFACE_MODE_GMII) {
 		ar8xxx_write(priv, AR8229_REG_OPER_MODE0,
-- 
2.20.1

